File: src/main/java/com/radixpro/enigma/be/persistency/BePersistencyInjector.java
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */

package com.radixpro.enigma.be.persistency;

import com.radixpro.enigma.be.persistency.mappers.BePersMappersInjector;
import com.radixpro.enigma.shared.converters.ShConvertersInjector;
import com.radixpro.enigma.ui.helpers.UiHelpersInjector;

public class BePersistencyInjector {

   private BePersistencyInjector() {
      // prevent instantiation
   }


   public static ChartDataDao injectChartDataDao() {
      return new ChartDataDao();
   }

   public static ConfigurationDao injectConfigurationDao() {
      return new ConfigurationDao();
   }

   public static DataFileDao injectDataFileDao() {
      return new DataFileDao(injectJsonReader(), BePersMappersInjector.injectInputDataSetMapper());
   }

   public static DataReaderCsv injectDataReaderCsv() {
      return new DataReaderCsv(ShConvertersInjector.injectCsv2LocationConverter(), UiHelpersInjector.injectDateTimeJulianCreator());
   }

   public static JsonReader injectJsonReader() {
      return new JsonReader();
   }

   public static JsonWriter injectJsonWriter() {
      return new JsonWriter();
   }

   public static PropertyDao injectPropertyDao() {
      return new PropertyDao();
   }

   public static VersionDao injectVersionDao() {
      return new VersionDao();
   }
}

File: src/main/java/com/radixpro/enigma/be/persistency/BePersistencyInjector.kt
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */
package com.radixpro.enigma.be.persistency

import com.radixpro.enigma.share.di.ShareInjector.injectJsonReader
import com.radixpro.enigma.statistics.di.StatsInjector.injectInputDataSetMapper
import com.radixpro.enigma.statistics.di.StatsInjector.injectStatsPathConstructor
import com.radixpro.enigma.statistics.di.StatsInjector.injectStatsProjMapper
import com.radixpro.enigma.statistics.persistency.GlobalDataDao
import com.radixpro.enigma.statistics.persistency.StatsProjDaoJson

object BePersistencyInjector {
    fun injectChartDataDao(): ChartDataDao {
        return ChartDataDao()
    }

    fun injectConfigurationDao(): ConfigurationDao {
        return ConfigurationDao()
    }

    fun injectDataFileDao(): GlobalDataDao {
        return GlobalDataDao(injectJsonReader(), injectInputDataSetMapper(), injectStatsPathConstructor())
    }

    fun injectStatsProjDao(): StatsProjDaoJson {
        return StatsProjDaoJson(injectJsonReader(), injectStatsProjMapper())
    }

    fun injectVersionDao(): VersionDao {
        return VersionDao()
    }
}